arm/ui/routes.py

```diff
@@ -185,15 +207,53 @@ def listlogs(path):
 @app.route('/')
 @app.route('/index.html')
 def home():
-    # freegb = getsize(cfg['RAWPATH'])
+    # Hard drive space
     freegb = psutil.disk_usage(cfg['ARMPATH']).free
-    freegb = round(freegb/1073741824, 1)
+    freegb = round(freegb / 1073741824, 1)
     mfreegb = psutil.disk_usage(cfg['MEDIA_DIR']).free
-    mfreegb = round(mfreegb/1073741824, 1)
+    mfreegb = round(mfreegb / 1073741824, 1)
+
+    # RAM memory
+    meminfo = dict((i.split()[0].rstrip(':'), int(i.split()[1])) for i in open('/proc/meminfo').readlines())
+    mem_kib = meminfo['MemTotal']  # e.g. 3921852
+    mem_gib = mem_kib / (1024.0 * 1024.0)
+    # lets make sure we only give back small numbers
+    mem_gib = round(mem_gib, 2)
+
+    memused_kib = meminfo['MemFree']  # e.g. 3921852
+    memused_gib = memused_kib / (1024.0 * 1024.0)
+    # lets make sure we only give back small numbers
+    memused_gib = round(memused_gib, 2)
+    memused_gibs = round(mem_gib - memused_gib, 2)
+
+    # get out cpu info
+    ourcpu = get_processor_name()
+
     if os.path.isfile(cfg['DBFILE']):
         # jobs = Job.query.filter_by(status="active")
         jobs = db.session.query(Job).filter(Job.status.notin_(['fail', 'success'])).all()
     else:
         jobs = {}
 
-    return render_template('index.html', freegb=freegb, mfreegb=mfreegb, jobs=jobs)
+    return render_template('index.html', freegb=freegb, mfreegb=mfreegb, jobs=jobs, cpu=ourcpu, ram=mem_gib,
+                           ramused=memused_gibs, ramfree=memused_gib, ramdump=meminfo)
+
+
+def get_processor_name():
+    if platform.system() == "Windows":
+        return platform.processor()
+    elif platform.system() == "Darwin":
+        return subprocess.check_output(['/usr/sbin/sysctl', "-n", "machdep.cpu.brand_string"]).strip()
+    elif platform.system() == "Linux":
+        command = "cat /proc/cpuinfo"
+        # return \
+        fulldump = str(subprocess.check_output(command, shell=True).strip())
+        # Take any float trailing "MHz", some whitespace, and a colon.
+        speeds = re.search(r"\\nmodel name\\t:.*?GHz\\n", fulldump)
+        # return str(fulldump)
+        speeds = str(speeds.group())
+        speeds = speeds.replace('\\n', ' ')
+        speeds = speeds.replace('\\t', ' ')
+        speeds = speeds.replace('model name :', '')
+        return speeds
+    return ""
```

Comment:
I had thought about something like this, but hesitated as I felt if I did CPU, I should probably do GPU utilization (nvidia-smi -l )

Reply:
The only reason i included the cpu is because it was easy to show what cpu the user is using no matter if they are using Intel or AMD.
This also helped for me personally, to make it clear if i was on my main servers page or my testing rig.

The only problem i see with adding GPU usage is that AMD GPU's and Intel APU's would not be supported. We have no idea on the percentages these are distributed in users. So its hard to know what percentage of users these changes would be useful too.

It is possible to very easily show CPU usage thanks to the already loaded psutil module. Its something i have already added in my own repo.